[Bug]dev install with npm 7.6 (idk the exact ver that breaks it)

**Did you read [the troubleshooting guide](https://hakuneko.download/docs/troubleshoot/)**
Yes

**Describe the bug**
`npm install` fails when using npm 7.6 or higher

**To Reproduce**
Steps to reproduce the behavior:
1. clone HN
2. `npm install` with 7.6 
3. it fails 

**Expected behavior**
works

**Screenshots**
see EvEr4EnGeL's installation on discord

**Context (please complete the following information):**
 - OS: [e.g. Windows10]
 - HakuNeko Version [e.g. 6.17]

**Additional context**
at least warn future devs by saying something in the [dev wiki](https://github.com/manga-download/hakuneko/wiki/Developer-Manual#3-install-dependencies)

ive had this issue come up on another persons install too (on non debian linux)

Quoting myself from the discord:

> Anyway, it's not a node 15/npm 7.6 issue, it's something else
> what i mean is that it's not about the binary themselves, it could be about something they now configure with the installer

The issue is on a "fresh install" (meaning the user never installed node/npm).
I'm using the latest (updated from a previous version) and i don't have the issue.

Issue was related to node-gyp being present and failing.

```
gyp ERR! find VS looking for Visual Studio 2015
npm ERR! gyp ERR! find VS - not found
npm ERR! gyp ERR! find VS not looking for VS2013 as it is only supported up to Node.js 8
```

The installer has a checkbox to install the "tools for native modules" (not checked  by default).
I think that i did check it in the past and that's probably the dependencies required.

--- a/package.json
+++ b/package.json
@@ -29,7 +29,7 @@
     "win-7zip": "latest"
   },
   "scripts": {
-    "postinstall": "cd src/app && npm install",
+    "postinstall": "cd src/app && npm install --legacy-peer-deps",
     "start": "npm run start:dev",
     "start:dev": "electron . --update-url=DISABLED --cache-directory=./src/web --disable-http-cache",
     "start:build": "npm run build:web && electron . --update-url=DISABLED --cache-directory=./build/latest",

--- a/package.json
+++ b/package.json
@@ -29,7 +29,7 @@
     "win-7zip": "latest"
   },
   "scripts": {
-    "postinstall": "cd src/app && npm install --legacy-peer-deps",
+    "postinstall": "cd src/app && npm install --no-optional",
     "start": "npm run start:dev",
     "start:dev": "electron . --update-url=DISABLED --cache-directory=./src/web --disable-http-cache",
     "start:build": "npm run build:web && electron . --update-url=DISABLED --cache-directory=./build/latest",

--- a/src/app/package.json
+++ b/src/app/package.json
@@ -11,8 +11,8 @@
   "main": "main.js",
   "dependencies": {
     "@logtrine/logtrine": "latest",
+    "discord-rpc": "^3.2.0",
     "fs-extra": "latest",
-    "jszip": "latest",
-    "discord-rpc": "^3.1.1"
+    "jszip": "latest"
   }
 }

